src/Umbraco.Web/Trees/FileSystemTreeController.cs

```diff
@@ -4,14 +4,46 @@
 using System.Net.Http.Formatting;
 using System.Web;
 using Umbraco.Core;
+using Umbraco.Core.Cache;
+using Umbraco.Core.Configuration;
 using Umbraco.Core.IO;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Mapping;
+using Umbraco.Core.Persistence;
+using Umbraco.Core.Services;
 using Umbraco.Web.Actions;
+using Umbraco.Web.Composing;
 using Umbraco.Web.Models.Trees;
+using Umbraco.Web.Routing;
 
 namespace Umbraco.Web.Trees
 {
     public abstract class FileSystemTreeController : TreeController
     {
+        private readonly IMenuItemCollectionFactory _menuItemCollectionFactory;
```

Comment:
Should all TreeControllers just expose this as a protected property?

Reply:
I change that - It is currently not used, but then we follow our pattern.